test/Test_Grammar.re

```diff
@@ -0,0 +1,26 @@
+open Alcotest;
+open Haz3lcore;
+open Grammar;
+
+let qcheck_map_annotation_test =
```

Comment:
Coverage guilting me into writing property tests. Future PR I'll move the generator to the Grammar types instead of the menhir ones.